Assets/Scripts/Auction/AuctionDriver.cs

```diff
@@ -41,38 +47,73 @@ private void Start()
+        if (MatchController.Singleton.RoundCount > 3)
+            availableAuctionStages = new WeightedRandomisedAuctionStage[] { StaticInfo.Singleton.BodyAuction, StaticInfo.Singleton.BarrelAuction, StaticInfo.Singleton.ExtensionAuction };
+
         playerFactory = GetComponent<PlayerFactory>();
         playerFactory.InstantiatePlayersBidding();
         playersInAuction = new HashSet<PlayerManager>(FindObjectsOfType<PlayerManager>());
 
         AnimateAuctionStart();
+        StartCoroutine(WaitAndStartCameraAnimation());
         StartCoroutine(PopulatePlatforms());
     }
 
+    private void OnDestroy()
+    {
+        for (int i = 0; i < biddingPlatforms.Length; i++)
+        {
+            biddingPlatforms[i].onBiddingExtended -= SetPrioritizedPlatform;
+            biddingPlatforms[i].onBidPlaced -= ActivateAuctioneerBid;
+            biddingPlatforms[i].onBiddingEnd -= ActivateAuctioneerSell;
+            biddingPlatforms[i].onBiddingExtended -= ActivateAuctioneerHaste;
+            biddingPlatforms[i].onBidDenied -= ActivateAuctioneerMissing;
+        }
+    }
+
     private void AnimateAuctionStart()
     {
+        //TODO: Animate auctioneer presenting items instead, meanwhile this functionality should still be here
         GlobalHUDController globalHUD = GetComponentInChildren<GlobalHUDController>();
-        StartCoroutine(globalHUD.DisplayStartScreen(biddingBeginDelay));
+        StartCoroutine(globalHUD.DisplayStartScreen(biddingBeginDelay+2));
+    }
+
+    private IEnumerator WaitAndStartCameraAnimation()
+    {
+        yield return new WaitForSeconds(biddingBeginDelay);
+        cameraAnimator.SetTrigger("start");
     }
 
     private IEnumerator PopulatePlatforms()
     {
         yield return new WaitForSeconds(biddingBeginDelay);
-        if (!(availableAuctionStages.Length == biddingPlatforms.Length))
-        {
-            Debug.Log("Not enough available auctionStages or biddingPlatforms!");
-        }
 
         lastExtendedAuction = biddingPlatforms[0];
         lastExtendedAuction.onBiddingEnd += EndAuction;
 
-        for (int i = 0; i < biddingPlatforms.Length; i++)
+        List<BiddingRound> biddingRounds = new List<BiddingRound>();
+        for (int i = 0; i < availableAuctionStages.Length; i++)
         {
             availableAuctionStages[i].Promote(out BiddingRound biddingRound);
-            biddingPlatforms[i].ActiveBiddingRound = biddingRound;
-            biddingPlatforms[i].SetItem(biddingRound.items[0]);
+            biddingRounds.Add(biddingRound);
+        }
+        bool isMultiple = availableAuctionStages.Length >= biddingPlatforms.Length;
+
+        for (int i = 0; i < biddingPlatforms.Length; i++)
+        {
+            biddingPlatforms[i].ActiveBiddingRound = biddingRounds[isMultiple ? i : 0];
+            biddingPlatforms[i].SetItem(biddingRounds[isMultiple ? i : 0].items[isMultiple ? 0 : i]);
```

Comment:
> So a bidding round is simply a series of item types?
Yep
> And we don't switch to other items after the platforms have been exhausted?
Kind of, items are replenished to the "pool" if every item has been exhausted in the selection, however, our item pools in staticInfo is sorted as a series of bodies, barrels and extensions. In order to "blend them" together in a specific order (We always want the auction to be ordered body, barrel, extension) and guarantee that the random selection is always done with one of each, I had to generate three biddingRounds with 1 item each instead of one biddinground with 3 items each. 

This is a bit convoluted, and could be simplified with a (large) refactor of the auctionstages. I considered doing so, but decided not to until we overhaul the BiddingPlatforms, which I decided to do later instead of in this PR. If that is right or not is of course up for debate 😅

Reply:
Aha. An auction stage refactor seems rather minor compared to other tasks though (like a refactor of the weapon system... and making mesh projectiles work (oh god))